### ISSUE TITLE
memory leack?

### ISSUE THREAD

**Issue by FengRongYue** [open-mmlab/mmdetection#10883]
![image](https://github.com/open-mmlab/mmdetection/assets/95663126/d505bff0-5ffc-448b-9337-5b7b7d89f95a)
i am traing deformable detr. i found the memery continuous rise. I try to down the num_works, but it dosent' work. My image resolution is big , 4000*4000. the batch size i used is 4.

**Comment by gongjizhang** [comment]
I meet same problem when train grounding dino。

**Comment by yewentao256** [comment]
While training the DETR model, I've noticed a consistent increase in memory usage over time. Could there potentially be a memory leak within this model's architecture or implementation?

**Comment by Xjh-UCAS** [comment]
Yes, I also found that when training DETR-related models, the CPU memory kept increasing, eventually causing the process to be killed.
